### ISSUE TITLE
Flakey test ClusteredSessionMigrationTest

### ISSUE THREAD

**Issue by gregw** [eclipse/jetty.project#4006]
I have seen the following error in CI twice.   Somehow it looks like the write on session create is being overtaken by the second request???

https://jenkins.webtide.net/job/jetty.project/job/jetty-9.4.x/1108/testReport/junit/org.eclipse.jetty.server.session/ClusteredSessionMigrationTest/Parallel_Stage___Build___Test___JDK8___testSessionMigration/

**Comment by joakime** [comment]
I've seen this in a few branch PR builds too.

https://jenkins.webtide.net/blue/organizations/jenkins/jetty.project/detail/PR-3995/9/tests/

```
Error:
expected: <200> but was: <500>

Stacktrace:
org.opentest4j.AssertionFailedError: expected: <200> but was: <500>
at org.eclipse.jetty.server.session.ClusteredSessionMigrationTest.testSessionMigration(ClusteredSessionMigrationTest.java:112)

Standard Error:
Running org.eclipse.jetty.server.session.ClusteredSessionMigrationTest.testSessionMigration()
2019-08-22 12:22:23.958:INFO:oejs.Server:main: jetty-9.4.21-SNAPSHOT; built: 2019-08-22T12:06:09.740Z; git: 74ff937ed37bd9f75c119da82eef29f4433aa8e8; jvm 1.8.0_222-b10
2019-08-22 12:22:23.960:INFO:oejs.session:main: DefaultSessionIdManager workerName=w2
2019-08-22 12:22:23.960:INFO:oejs.session:main: w2 Scavenging every 11000ms
2019-08-22 12:22:24.206:INFO:oejsh.ContextHandler:main: Started o.e.j.s.ServletContextHandler@7451338e{/,null,AVAILABLE}
2019-08-22 12:22:24.207:INFO:oejs.AbstractConnector:main: Started ServerConnector@43c76550{HTTP/1.1,[http/1.1]}{0.0.0.0:36995}
2019-08-22 12:22:24.207:INFO:oejs.Server:main: Started @10085ms
2019-08-22 12:22:24.207:INFO:oejs.Server:main: jetty-9.4.21-SNAPSHOT; built: 2019-08-22T12:06:09.740Z; git: 74ff937ed37bd9f75c119da82eef29f4433aa8e8; jvm 1.8.0_222-b10
2019-08-22 12:22:24.226:INFO:oejs.session:main: DefaultSessionIdManager workerName=w3
2019-08-22 12:22:24.226:INFO:oejs.session:main: w3 Scavenging every 11000ms
2019-08-22 12:22:24.268:INFO:oejsh.ContextHandler:main: Started o.e.j.s.ServletContextHandler@7cb7ca06{/,null,AVAILABLE}
2019-08-22 12:22:24.269:INFO:oejs.AbstractConnector:main: Started ServerConnector@5cfb4b71{HTTP/1.1,[http/1.1]}{0.0.0.0:33421}
2019-08-22 12:22:24.269:INFO:oejs.Server:main: Started @10147ms
2019-08-22 12:22:24.299:WARN:oejs.HttpChannel:qtp210949780-60: /server
java.lang.NullPointerException
at org.eclipse.jetty.server.session.ClusteredSessionMigrationTest$TestServlet.doPost(ClusteredSessionMigrationTest.java:160)
at org.eclipse.jetty.server.session.ClusteredSessionMigrationTest$TestServlet.doGet(ClusteredSessionMigrationTest.java:139)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:852)
at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:544)
at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1568)
at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1307)
at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:482)
at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1538)
at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1204)
at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:221)
at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
at org.eclipse.jetty.server.Server.handle(Server.java:494)
at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:374)
at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:268)
at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:782)
at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:918)
at java.lang.Thread.run(Thread.java:748)
2019-08-22 12:22:24.315:INFO:oejs.AbstractConnector:main: Stopped ServerConnector@5cfb4b71{HTTP/1.1,[http/1.1]}{0.0.0.0:0}
2019-08-22 12:22:24.319:INFO:oejsh.ContextHandler:main: Stopped o.e.j.s.ServletContextHandler@7cb7ca06{/,null,UNAVAILABLE}
2019-08-22 12:22:24.319:INFO:oejs.session:main: w3 Stopped scavenging
2019-08-22 12:22:24.321:INFO:oejs.AbstractConnector:main: Stopped ServerConnector@43c76550{HTTP/1.1,[http/1.1]}{0.0.0.0:0}
2019-08-22 12:22:24.324:INFO:oejsh.ContextHandler:main: Stopped o.e.j.s.ServletContextHandler@7451338e{/,null,UNAVAILABLE}
2019-08-22 12:22:24.325:INFO:oejs.session:main: w2 Stopped scavenging
```

**Comment by janbartel** [comment]
Fixed for 9.4.21 and beyond.
